server/remote_cache/action_cache_server/action_cache_server.go

```diff
@@ -28,7 +28,33 @@ import (
 	rspb "github.com/buildbuddy-io/buildbuddy/proto/resource"
 )
 
-var checkClientActionResultDigests = flag.Bool("cache.check_client_action_result_digests", false, "If true, the server will check (and honor) the bb-specific cached_action_result_digest field on ActionCache.getActionResult requests to reduce bandwidth")
+var (
+	checkClientActionResultDigests = flag.Bool("cache.check_client_action_result_digests", false, "If true, the server will check (and honor) the bb-specific cached_action_result_digest field on ActionCache.getActionResult requests to reduce bandwidth")
+
+	restrictedPrefixes []string
+	restrictedOnce     sync.Once
+	restrictedMu       sync.Mutex
+)
+
+// RestrictPrefix adds an instance name prefix to a list of prefixes that are "restricted":
+// the Action Cache Server will not return an Action Result stored under a restricted instance name
+// prefix unless the request comes from the app, an executor, or a workflow.
+// RestrictPrefix must be only called in init() functions, so that isRestricted(...) does not
+// have to lock a mutex during Action Cache request serving.
+func RestrictPrefix(prefix string) {
```

Comment:
I think it's fine to not make it so general for now. You can create a variable with a hardcoded list of prefixes, something like:
```
var (
	appRestrictedPrefixes = []string{interfaces.OCIArtifactInstanceNamePrefix}
)
``` 
and then in the check validate that the request is coming from the apps (don't need to allow executors or workflows for now).

If we need something more general it will be very easy to change in the future.

Reply:
Thanks, makes life a lot easier.